### DISCUSSION TITLE
1.4.2

### DISCUSSION THREAD

**Discussion by BoloVEVO** [BoloVEVO/Kade-Engine#10] [Announcements]
### **Fixes:**
- Fixed Switching States memory leak
- Fixed NoteSplashes not matching with note colors when Note Color quantization is enabled.
- Fixed precaching null exceptions.

### **Features:**
- **New Option: GPU Rendering:** If enabled (by default) it pushes the sprite load to Graphic card instead of RAM to decrease memory usage.
- **New Option: HitSound mode** (On Note Hit or On Strum Hit)
- **Customize Gameplay Zoom:** Dynamic HUD zoom, now zooming will scale HUD objects height to give a more comfortable view to user (for those who like quaver hud layout).
- **Customize Gameplay Appareance section:** A little hud tab to enable/disable some gameplay modules such as combo numbers, combo sprite and miliseconds text.
- **Better Optimization:** Now the game handles memory in a smart way to not make your computer explode 😈
- **Individual receptor note direction:** You can now do sick modcharts changing the note scroll direction.

### **Misc:**
- **Engine Watermark Option:** If disabled the game will play the original FNF main menu music.
- **Better Custom Difficulties Support:** Now it should be easy to make custom difficulties without getting any null exceptions while messing in source code. This means you can make custom difficulties for some songs. (In Story Mode too)
- **Better Custom Charts and Characters Support:** Now you can drag custom charts and characters (json and assets) to executable kade engine and play it in freeplay mode. (Custom song charts need to have at least one FNF difficulty otherwise it'll boom)
- **Paths:** For cpp targets, the game will use FileSystem to acess to assets. Otherwise, it will use OpenFLAssets as usual. (This doesn't affect anything to Paths functions btw)
- **Characters AtlasFramesMaker Support:** Now you can use characters that use AtlasFrames instead of Sparrow or PacketAtlas. (Need to be specified in character's json!)

### **Known Issues**
- HTML5 Request Animation high latency.
- ChartingState and Stage Debug very buggy.

College is raping rn. I think this is the definitive edition of Kade for now. Cya 🗿

<hr /><em>This discussion was created from the release <a href='https://github.com/BoloVEVO/Kade-Engine-Public/releases/tag/v1.4.2'>1.4.2</a>.</em>

**Comment by Shocka12** [comment]
Whats would be in next update? Random chart mode? Combo Breaks limit mode? Or debug mode in menu(like in psych engine) Im curious. 
Good Luck on working!

**Reply by BoloVEVO** [reply]
Yes, next release is focused in editors and gameplay fixes and optm.

**Reply by Shocka12** [reply]
I hope the cutscenes in week 7 will be skippable(restart song in story mode) in next update. 
And i have 3 little questions:
1. The custom difficulties will be easier to add for next update?
2. The character offsetts will be fixed?(like a psych engine)
3. Whats your favorite mod?

**Reply by BoloVEVO** [reply]
Custom difficulties are now easy to implement via source code in CoolUtil.hx just adding them to the array and suffix array respectively.

I'm not sure what is wrong with chars offset, so I can't answer you srry lol

My fav mod is vs Camellia (ironically I'm the lead coder there xD)

**Comment by Shocka12** [comment]
Thanks for answers! Im bad at english. I mean the character position in stages.  Sorry for confusing.
![image](https://user-images.githubusercontent.com/122198793/226378332-e02be732-fe48-4734-bc5e-27af39833a40.png)

**Reply by BoloVEVO** [reply]
Ah it's because stages handles characters' position depending of what characters are in the stage.

For example you set Pico and BF position in week 3 but when you switch from pico to daddy dearest, he's in a completely different position because he's not pico.


Check Stage.hx there's a positions variable where you can set a unique position for each character using their json names (the name you set to the char json ex: bf-holding-gf.json the name will be bf-holding-gf )

**Reply by Shocka12** [reply]
Thanks!
![x53SG9vKbdY](https://user-images.githubusercontent.com/122198793/226385498-fad7e3d6-c604-45fe-b4ef-9ea09f11aee7.jpg)

**Comment by Shocka12** [comment]


**Reply by BoloVEVO** [reply]
post it on issues this chat is not for issues pal.
